src/main/java/com/teragrep/jai_02/keystore/KeyStoreAccess.java

```diff
@@ -0,0 +1,169 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://github.com/teragrep/teragrep/blob/main/LICENSE>.
+ *
+ *
+ * Additional permission under GNU Affero General Public License version 3
+ * section 7
+ *
+ * If you modify this Program, or any covered work, by linking or combining it
+ * with other code, such other code is not for that reason alone subject to any
+ * of the requirements of the GNU Affero GPL version 3 as long as this Program
+ * is the same Program as licensed from Suomen Kanuuna Oy without any additional
+ * modifications.
+ *
+ * Supplemented terms under GNU Affero General Public License version 3
+ * section 7
+ *
+ * Origin of the software must be attributed to Suomen Kanuuna Oy. Any modified
+ * versions must be marked as "Modified version of" The Program.
+ *
+ * Names of the licensors and authors may not be used for publicity purposes.
+ *
+ * No rights are granted for use of trade names, trademarks, or service marks
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jai_02.keystore;
+
+import javax.crypto.SecretKey;
+import java.io.IOException;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.UnrecoverableEntryException;
+import java.security.spec.InvalidKeySpecException;
+import java.util.*;
+
+/**
+ * Provides access to the KeyStore, such as loading, saving
+ * and deleting entries. Keeps track of the username->alias mapping
+ * via the UserToAliasMapping object.
+ */
+public class KeyStoreAccess {
+    private final String keyStorePath;
+    private final char[] keyStorePassword;
+    private final EntryAliasFactory entryAliasFactory;
+    private final UserToAliasMapping userToAliasMapping;
+    private final KeyStore keyStore;
+
+    public KeyStoreAccess(final String keyStorePath, final char[] keyStorePassword) {
+        this(keyStorePath, keyStorePassword, new EntryAliasFactory());
+    }
+    public KeyStoreAccess(final String keyStorePath, final char[] keyStorePassword, final EntryAliasFactory entryAliasFactory) {
+        this.entryAliasFactory = entryAliasFactory;
+        this.keyStorePassword = keyStorePassword;
+        this.keyStorePath = keyStorePath;
+        this.keyStore = new KeyStoreFactory(keyStorePath, keyStorePassword).build();
```

Comment:
is factory necessary here or could it use just KeyStore as an argument?

Reply:
The reason factory is used is to produce a ready to use KeyStore. This could be removed, however the KeyStoreFactory makes sure only known good keyStoreFormats are used, the given path is checked and the keyStore is initialized and pre-existing data loaded.

Reply:
I think it should be either encapsulated into a separate object if we want to restrict it to certain specification or just decouple it, it's up to the user to decide after all which format to use